### PR TITLE
pass time duration instead of 'reconcile.Result' through functions in adv ds

### PR DESCRIPTION
**Pull request by dylandee**
`reconcile.Request` is passing through methods in adv ds just for delay time duration. However `reflect.DeepEqual` uses reflect package which is low efficiency.

So I think its better to just pass delay time directly.

### CODE REVIEW

File: pkg/controller/daemonset/daemonset_controller.go [hunk 1]
```diff
@@ -367,26 +367,27 @@ func (dsc *ReconcileDaemonSet) syncDaemonSet(request reconcile.Request) (reconci
...
 	}
 
-	result, err := dsc.manage(ds, hash)
-	if err != nil || !reflect.DeepEqual(result, reconcile.Result{}) {
-		return result, err
+	delay, err := dsc.manage(ds, hash)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if int(delay) != 0 {
```
Comment (Fei-Guo) [inline pkg/controller/daemonset/daemonset_controller.go:374]: Do you need to convert to int type?

File: pkg/controller/daemonset/daemonset_controller.go [hunk 2]
```diff
@@ -367,26 +367,27 @@ func (dsc *ReconcileDaemonSet) syncDaemonSet(request reconcile.Request) (reconci
...
-				if !reflect.DeepEqual(result, reconcile.Result{}) {
-					klog.V(6).Infof("delay to requeue.: %v", result)
-					return result, err
-				}
+			delay, err = dsc.rollingUpdate(ds, hash)
+			if err != nil {
+				klog.Errorf("rollingUpdate failed: %v", err)
+				return reconcile.Result{}, err
+			}
+			if int(delay) != 0 {
```
Comment (Fei-Guo) [inline pkg/controller/daemonset/daemonset_controller.go:388]: Do you need int conversion?

File: pkg/controller/daemonset/daemonset_controller.go [hunk 3]
```diff
@@ -601,13 +601,12 @@ func (dsc *ReconcileDaemonSet) manage(ds *appsv1alpha1.DaemonSet, hash string) (
...
 		}
-		re, nodesNeedingDaemonPodsOnNode, podsToDeleteOnNode, err := dsc.podsShouldBeOnNode(node, nodeToDaemonPods, ds, hash)
+		delay, nodesNeedingDaemonPodsOnNode, podsToDeleteOnNode, err := dsc.podsShouldBeOnNode(node, nodeToDaemonPods, ds, hash)
 		if err != nil {
 			continue
 		}
-
-		if !reflect.DeepEqual(re, reconcile.Result{}) {
-			return re, nil
+		if int(delay) != 0 {
```
Comment (Fei-Guo) [inline pkg/controller/daemonset/daemonset_controller.go:608]: same here.

File: pkg/controller/daemonset/daemonset_controller.go [hunk 4]
```diff
@@ -805,7 +799,9 @@ func (dsc *ReconcileDaemonSet) podsShouldBeOnNode(
 	nodeToDaemonPods map[string][]*corev1.Pod,
 	ds *appsv1alpha1.DaemonSet,
 	hash string,
-) (result reconcile.Result, nodesNeedingDaemonPods, podsToDelete []string, err error) {
+) (duration time.Duration, nodesNeedingDaemonPods, podsToDelete []string, err error) {
+
+	var requeueDuration = requeueduration.Duration{}
```
Comment (Fei-Guo) [inline pkg/controller/daemonset/daemonset_controller.go:804]: Why do you use requeueduration.Duration here? I don't see concurrent access issue in this method.

**Review by Fei-Guo — COMMENTED** [review]
Thanks for working on this. I think this is a reasonable cleanup. Comments below.

File: pkg/controller/daemonset/daemonset_controller.go (hunk 4 above)
Comment (dylandee) [reply pkg/controller/daemonset/daemonset_controller.go:804]: > Why do you use requeueduration.Duration here? I don't see concurrent access issue in this method.

I just find it convenient to update time duarion with requeueduration.Duration,  i will fix it.

File: pkg/controller/daemonset/daemonset_controller.go (hunk 1 above)
Comment (dylandee) [reply pkg/controller/daemonset/daemonset_controller.go:374]: yeah, its redundant

**Review by Fei-Guo — APPROVED** [review]
LGTM.

Thanks for the clean up.

**Pull request merged**